### DISCUSSION TITLE
Is there a Public Roadmap?

### DISCUSSION THREAD

**Discussion by Evalprime** [bluesky-social/atproto#1147] [General Discussion]
is there a public roadmap for upcoming features? if not, could we use GitHub projects?

**Comment by bnewbold** [comment]
We don't have a formal public roadmap, and we intentionally try not to publicly commit to specific dates for specific features. We think this is the healthy and correct thing to do for most software projects!

There was a bit of discussion of upcoming priorities in our "Private Beta Update" blog post: https://blueskyweb.xyz/blog/6-02-2023-beta-update

In hand-wave-y terms:

* moderation tooling and features are the current top priority, and block launching federation
* getting to public federation of our production service (bsky.social) is the big milestone. we will probably do sandbox networks sooner to let folks explore and poke at the protocol in a lower-stakes environment. the architecture, protocol, and implementation work is mostly done, but there are many details to work out to ensure everything goes smoothly
* bugs, scaling, and security fixes frequently pre-empt other work and get shipped quickly, as you might expect
* documentation and specification work is going in parallel with other work
* things like DMs, non-public data, or significant new application features are all very unlikely to be even planned out until after launching open federation

**Reply by Gargaj** [reply]
How about a roadmap in terms of core changes that come from design/UX? Or rather, is there a plan to at some point deviate away from the current Twitter feature set?

What I mean is something like "let's have Slack-style reactions instead of a single like button" or "let's allow Instagram-style co-authored posts" or "let's decide between 1 or more pinned posts, cf. Twitter v Mastodon" - these are decisions that aren't specific to the federation protocol, but are specific to the BSky app on top of the protocol.

**Reply by bnewbold** [reply]
The product team has lots and lots of ideas, but I don't think we can give you many specifics on features or timelines. There will probably be more time and headspace to work on design and product features once the current round of federation and composable moderation work is complete.
